### ISSUE TITLE
kvserver: `TestPromoteNonVoterInAddVoter` flaked on `release-23.1`

### ISSUE THREAD

**Issue by erikgrinaker** [cockroachdb/cockroach#105883]
Seen in a [CI failure](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/10734994?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true).

```
=== RUN   TestPromoteNonVoterInAddVoter
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestPromoteNonVoterInAddVoter2438547430
    test_log_scope.go:79: use -show-logs to present logs inline
    replicate_queue_test.go:2248: condition failed to evaluate within 3m45s: Range counts are not all equal
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestPromoteNonVoterInAddVoter2438547430
--- FAIL: TestPromoteNonVoterInAddVoter (462.14s)
```

Jira issue: CRDB-29255

**Comment by kvoli** [comment]
Seems different from https://github.com/cockroachdb/cockroach/issues/101519. I should open a backport for that anyway.

**Comment by kvoli** [comment]
This will be inherently flaky under race/stress. The test requires the range count to be balanced initially. This isn't the behavior the test is interested in.

Opened a PR to skip under race.

**Comment by kvoli** [comment]
> @kvoli don't we need this on master as well? I just hit this on `master` here: [teamcity.cockroachdb.com/viewLog.html?buildId=10865562&tab=buildResultsDiv&buildTypeId=Cockroach_Ci_Tests_Testrace](https://teamcity.cockroachdb.com/viewLog.html?buildId=10865562&tab=buildResultsDiv&buildTypeId=Cockroach_Ci_Tests_Testrace)

https://github.com/cockroachdb/cockroach/pull/106514 is the master PR. I'll merge today.

**Comment by kvoli** [comment]
Will be closed on https://github.com/cockroachdb/cockroach/pull/106628

**Comment by erikgrinaker** [comment]
Resolved by #106628.
